### PR TITLE
Tests for jf-list-maker and jf-billboard

### PR DESCRIPTION
**Pull request by vivekKodira**

### CODE REVIEW

File: src/components/JfListMakerComponent.spec.js [hunk 1]
```diff
@@ -1,50 +1,389 @@
...
+localVue.directive('jf-tooltip-on-overflow', {
+    bind: function() {}
+});
+
+// Utility method to avoid code duplication
+const mountComponent = function( props ){
+    let options = {
+        localVue,
+        propsData: Object.assign({}, props),
+        stubs: ['jf-field','jf-ui-select']
```
Comment (ur92) [inline src/components/JfListMakerComponent.spec.js]: when you shallowMount, vue will stub automatically all child components. You can consider to remove manual stubs.

File: src/components/JfListMakerComponent.spec.js [hunk 2]
```diff
@@ -1,50 +1,389 @@
...
+const MSG_MUST_INPUT_VALUE = `Must input value`;
+const MSG_VALUE_ALREADY_EXISTS = `Value already exists`
+
 describe('JfListMakerComponent', () => {
-    it('renders a form', () => {
-        const wrapper = shallowMount(JfListMakerComponent, {
-            propsData: {},
-            stubs: ['jf-field']
+    describe("display", function() {
+        const wrapper = mountComponent();
```
Comment (ur92) [inline src/components/JfListMakerComponent.spec.js]: You have to wrap the setup in beforeEach as currently is you setup once and before each test, your tests are not clean and effecting each other.

File: src/components/JfListMakerComponent.spec.js [hunk 3]
```diff
@@ -1,50 +1,389 @@
...
+//Messages
+const MSG_MUST_INPUT_VALUE = `Must input value`;
+const MSG_VALUE_ALREADY_EXISTS = `Value already exists`
+
 describe('JfListMakerComponent', () => {
-    it('renders a form', () => {
-        const wrapper = shallowMount(JfListMakerComponent, {
-            propsData: {},
-            stubs: ['jf-field']
+    describe("display", function() {
```
Comment (ur92) [inline src/components/JfListMakerComponent.spec.js]: excellent nested describes breakage!

File: src/components/JfListMakerComponent.spec.js [hunk 4]
```diff
@@ -1,50 +1,389 @@
...
+    let options = {
+        localVue,
+        propsData: Object.assign({}, props),
+        stubs: ['jf-field','jf-ui-select']
+    }
+    return shallowMount(JfListMakerComponent, options);
+}
+
+//selectors
+const SELECTOR_A_LIST_ROW = `.list-maker-row-value`;
```
Comment (ur92) [inline src/components/JfListMakerComponent.spec.js]: Good idea, Ill adopt this practice

File: src/components/JfListMakerComponent.spec.js [hunk 5]
```diff
@@ -1,50 +1,389 @@
...
+                //One event contains the new value entered
+                expect(
+                    wrapper.emitted()[EVENT_ON_ADD_VALUE][0]
+                ).toEqual([{ newValue: 'b' }])
+                // One event contains the updated array (necessary for two-way binding)
+                expect(wrapper.emitted()[EVENT_INPUT][0]).toEqual([
+                    ['a', 'b'],
+                ])
+                expect(
+                    wrapper.emitted()[EVENT_ON_AFTER_ADD_VALUE]
```
Comment (ur92) [inline src/components/JfListMakerComponent.spec.js]: Is the 'input' event bound to some method in the component? If yes, you can mock this method with `jest.fn()` and then you can expect if method called and sample the args.

File: src/components/JfListMakerComponent.spec.js [hunk 6]
```diff
@@ -1,50 +1,389 @@
...
+            //delete button
+            expect(
+                wrapper
+                    .contains(
+                        BUTTON_DELETE
+                    )
+            ).toBe(true)
+        })
+
+        it('should display new values in the list', function () {
```
Comment (ur92) [inline src/components/JfListMakerComponent.spec.js]: When you do mainly html tree checks, you can use jest's snapshot testing. So you can mount and snapshot the component, very useful and 'cheap' to write IMHO.
https://vue-test-utils.vuejs.org/guides/#snapshot-testing

File: src/components/JfListMakerComponent.spec.js (hunk 1 above)
Comment (vivekKodira) [reply src/components/JfListMakerComponent.spec.js]: @ur92 it doesn't do so for components registered as global

File: src/components/JfListMakerComponent.spec.js (hunk 2 above)
Comment (vivekKodira) [reply src/components/JfListMakerComponent.spec.js]: ah! you're right. will do.

File: src/components/JfListMakerComponent.spec.js (hunk 5 above)
Comment (vivekKodira) [reply src/components/JfListMakerComponent.spec.js]: The test is validating if component emits the 'input' event.

File: src/components/JfListMakerComponent.spec.js (hunk 6 above)
Comment (vivekKodira) [reply src/components/JfListMakerComponent.spec.js]: Thank you. Will add this for jf-billboard tests

File: src/components/JfListMakerComponent.spec.js (hunk 1 above)
Comment (ur92) [reply src/components/JfListMakerComponent.spec.js]: Oh interesting! Didn't know that... That could explain few issues that I have.

**Pull request merged**